File: server/src/readjson/Buff.java
package readjson;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class Buff implements Serializable {
    protected String name;
    protected String durationType;
    private HashMap<Integer, Integer> duration;
    protected Boolean durationUseCardLevel = false;
    protected Boolean effectsUseCardLevel = false;
    protected String state;

    protected HashMap<Integer, ArrayList<EffectBuff>> effects;

    public double getDuration(int evolution, int level) {
        double duration = this.duration.get(evolution) / 1000.0;
        if (this.durationUseCardLevel)
            duration *= 1 + 0.1 * (level - 1);
        return duration;
    }

    public double getEffectValue(int evolution) {
        return this.effects.get(evolution).get(0).value;
    }

    public String getEffectName(int evolution) {
        return this.effects.get(evolution).get(0).name;
    }
}

class EffectBuff implements Serializable {
    String name;
    String type;
    double value;
    double delay;
    int hpPerDelay;
}

File: client/src/modules/readjson/Buff.js
/**
 * Created by Team 2 - LongLH - GDF 17 on 12/12/2022.
 */


var Buff = cc.Class.extend({

    name: String.prototype,
    durationType: String.prototype,
    duration: Map.prototype,
    durationUseCardLevel: false,
    effectsUseCardLevel: false,
    state: String.prototype,

    effects: Map.prototype,

    getDuration: function (evolution, level) {
        let duration = this.duration[evolution] / 1000;
        if (this.durationUseCardLevel)
            duration *= 1 + 0.1 * (level - 1);
        return duration;
    },

    getEffectValue: function (evolution) {
        return this.effects[evolution][0].value;
    },

    getEffectName: function (evolution) {
        return this.effects[evolution][0].name;
    }

});

var EffectBuff = cc.Class.extend({

    name: String.prototype,
    type: String.prototype || Number.prototype,
    value: Number.prototype,
    delay: Number.prototype,
    hpPerDelay: Number.prototype,

});